src/containers/Tenant/Diagnostics/Partitions/Partitions.tsx

```diff
@@ -43,215 +40,162 @@ export const b = block('ydb-diagnostics-partitions');
-
+export const Partitions = ({path}: PartitionsProps) => {
     const dispatch = useDispatch();
 
-    const [generalSearchValue, setGeneralSearchValue] = useState('');
-    const [partitionIdSearchValue, setPartitionIdSearchValue] = useState('');
-
+    // Manual path control to ensure it updates with state reset
+    // so no request with wrong params will be sent
     const [componentCurrentPath, setComponentCurrentPath] = useState(path);
 
-    const {autorefresh} = useTypedSelector((state) => state.schema);
-    const {loading, wasLoaded, error, selectedConsumer} = useTypedSelector(
-        (state) => state.consumer,
-    );
+    // Initial content, data depends on entity type, it's different for partitions with and without consumers
+    const [partitions, setPartitions] = useState<PreparedPartitionDataWithHosts[]>([]);
+    const [columns, setColumns] = useState<Column<PreparedPartitionDataWithHosts>[]>([]);
+    const [columnsIdsForSelector, setColumnsIdsForSelector] = useState<string[]>([]);
 
-    const partitions = useTypedSelector((state) => selectPreparedPartitionsData(state));
+    // Data after applying filters
+    const [dataToRender, setDataToRender] = useState<PreparedPartitionDataWithHosts[]>([]);
 
-    const savedSelectedColumns: string = useTypedSelector((state) =>
-        getSettingValue(state, PARTITIONS_SELECTED_COLUMNS_KEY),
+    // Shallow equal ensures no rerender loops
+    const consumers = useTypedSelector(selectConsumersNames, shallowEqual);
+    
+    const nodesMap = useTypedSelector(selectNodesMap);
+    const partitionsWithConsumers = useTypedSelector(selectPreparedPartitionsData);
+    const partitionsWithoutConsumers = useTypedSelector(selectPreparedTopicPartitions);
+
+    const {autorefresh} = useTypedSelector((state) => state.schema);
+    const {
+        loading: consumerLoading,
+        wasLoaded: consumerWasLoaded,
+        error: consumerError,
+        selectedConsumer,
+    } = useTypedSelector((state) => state.consumer);
+    const {
+        loading: topicLoading,
+        wasLoaded: topicWasLoaded,
+        error: topicError,
+    } = useTypedSelector((state) => state.topic);
+    const {
+        loading: nodesLoading,
+        wasLoaded: nodesWasLoaded,
+        error: nodesError,
+    } = useTypedSelector((state) => state.nodesList);
+
+    const hiddenColumns: string[] = useTypedSelector(
+        (state) => getParsedSettingValue(state, PARTITIONS_HIDDEN_COLUMNS_KEY) || [],
     );
 
     useEffect(() => {
-        // Manual path control to ensure it updates with other values so no request with wrong params will be sent
+        dispatch(cleanTopicData());
+        dispatch(setTopicDataWasNotLoaded());
+
+        dispatch(getNodesList());
+        dispatch(getTopic(path));
+
         setComponentCurrentPath(path);
     }, [dispatch, path]);
 
-    const fetchConsumerData = useCallback(
+    // Data, its source and presentation is different for partitions with and without consumers
```

Comment:
is it possible to split logic into two custom hooks - handle data with and without consumers?

Reply:
As for it won’t make it easier, as there is a problem - there still should be some logic that will determine what entity we are dealing with

Reply:
Yes, but now this logic is splitted between useEffect and fetchData. My point was that you pass `selectedConsumer` into the hook and it returns `fetchData, columns, columnsIds` to isolate logic connected with difference you mentioned.